---
title: "iPSC to cardiomyocyte differentiation timecourse"
subtitle: "lncMYH7b KD compared to control iPSCs"
author: "Michael Smallegan"
date: "9/23/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(rtracklayer)
library(janitor)
library(ggrepel)
library(pheatmap)
library(ggpubr)
source("../../util/_plot_theme.R")
```

```{r data-import, include=FALSE}
samples <- read_csv("../diff_samplesheet.csv") %>%
  clean_names() %>%
  # Make some factors out of the relevant variables.
  mutate(condition = factor(condition, levels = c("CTL", "myh7b_KD")),
         timepoint = factor(timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))) %>% 
  # Put the columns in order
  arrange(condition, timepoint, bio_replicate) %>%
  # Add a more meaningful name for plots
  unite(sample_name, timepoint, condition, bio_replicate, remove = FALSE) %>%
  mutate(sample_name = gsub("myh7b_", "", sample_name))

# Grab the counts
salmon_counts <- read_csv("../../results/salmon/salmon_merged_gene_counts.csv") %>%
  column_to_rownames("gene_id") %>%
  as.matrix()

# Filter genes with less than 10 counts across the whole dataset
salmon_counts <- salmon_counts[rowSums(salmon_counts) > 10,]

# Get the annotations
# gtf <- rtracklayer::import("../../util/gencode.v33.annotation.gtf")
# genes <- gtf[gtf$type == "gene"]
# rtracklayer::export(genes, "results/gencode_v33_genes.gtf")
genes <- rtracklayer::import("results/gencode_v33_genes.gtf")
# g2s <- genes %>% 
#   as.data.frame() %>%
#   dplyr::select(gene_id, gene_name)
# write_csv(g2s, "results/gene2symbol.csv")
g2s <- read_csv("results/gene2symbol.csv")

# Line it all up
names(genes) <- genes$gene_id
genes <- genes[rownames(salmon_counts)]
stopifnot(all(names(genes) == rownames(salmon_counts)))

samples <- samples %>% as.data.frame()
rownames(samples) <- samples$sample_id
salmon_counts <- salmon_counts[,rownames(samples)]
stopifnot(all(rownames(samples) == colnames(salmon_counts)))

# Change count matrix to integer mode
mode(salmon_counts) <- "integer"
```

# lncMYH7b KD during cardiomyocyte differentiation

We observed that lncMYH7b is required for iPSCs to properly
differentiate into cardiomyocytes. Comparing the transcriptome over the
course of differentiation from lncMYH7b KD iPSCs to control iPSCs can
answer a few key questions:

1.  Globally, how well do the transcriptomes of lncMYH7b KD cells and
    control cells concord over the differentiation timecourse?

2.  At what stage of differentiation do lncMYH7b KD cells start to
    diverge from controls?

3.  To what extent do the genes involved in lineage commitment diverge
    from controls?

4.  Are lncMYH7b KD cells "stuck" in the middle of lineage commitment or
    do they start to look like an entirely different cell type?

5.  Are there steps early in the differentiation process that are
    misregulated by lncMYH7b depletion?

6.  Since other results indicate that the WNT signalling pathway may be
    implicated:

7.  Is the WNT signalling pathway affected?

8.  Do gene expression changes give us an insight into how the pathway
    is affected?

9.  In the final stages of differentiation, are there genes or pathways
    that contribute to explaining the observed phenotype?

## Transcriptome concordance over differentiation

Here we will take a look at the mean expression across replicates for
each timepoint and condition in a PCA plot and heatmap to determine to
what extent the lncMYH7b cells are following the same differentiation
pathway and how closely their transcriptomes end up resembling control
cardiomyocytes.

```{r transcriptome-concordance, echo = FALSE, message = FALSE, warning=FALSE, fig.height = 550px}
# Here we will use the regularized log normalized counts from DESeq2
# For that we need to make a DESeqDataSet object.
dds <- DESeqDataSetFromMatrix(countData = salmon_counts,
                              colData = samples,
                              design = ~ timepoint + condition + timepoint*condition,
                              rowData = genes)
# * commented out for speed in knitting * #
# rld <- rlog(dds, blind = FALSE)
# write_rds(rld, "results/rld_counts.rds")
rld <- read_rds("results/rld_counts.rds")
rld_matrix <- assay(rld)

# PCA plot with the mean normalized count across each sample.
# For this we will take just the genes which have high variance, 
# which is more informative for the trajectory of the transcriptomes.
# Calculate the variance across each gene. 
gene_var <- rowVars(rld_matrix)
# Take genes with variance in the 75th-percentile and up 
rld_var <- rld_matrix[gene_var >= quantile(gene_var, 0.9),]

# Merge replicates based on the mean normalized counts across the replicates.
rld_df <- rld_var %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>%
  pivot_longer(2:ncol(.), names_to = "sample_id", values_to = "rld") %>%
  left_join(samples) %>%
  unite(tcondition, condition, timepoint) %>%
  group_by(gene_id, tcondition) %>%
  summarize(mean_rld = mean(rld)) %>%
  pivot_wider(names_from = tcondition, values_from = mean_rld) %>%
  column_to_rownames("gene_id") %>%
  as.matrix()

# PCA
pca <- prcomp(rld_df, rank = 2)

# Get the proportion of variance explained by each principle component.
pca_summary <- summary(pca)$importance
prop_variance <- round(pca_summary["Proportion of Variance", 1:2] * 100)

# Collect into a data.frame for plotting
pca_df <- pca$rotation %>% 
  as.data.frame() %>%
  rownames_to_column("tcondition") %>%
  mutate(condition = ifelse(grepl("CTL", tcondition), "ctl", "kd"),
         timepoint = gsub("CTL_|myh7b_KD_", "", tcondition))
pca_df$timepoint <- factor(pca_df$timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))
pca_df$condition <- factor(pca_df$condition, levels = c("ctl", "kd"))

## Make a figure
differentiation_pca <- ggplot(pca_df, aes(x = PC1, y = PC2, color = condition, label = timepoint)) +
  geom_point(size = 3) +
  geom_text_repel() +
  theme_paperwhite() +
  xlab(paste0("PC1: ", prop_variance[[1]], "% variance")) +
  ylab(paste0("PC2: ", prop_variance[[2]], "% variance")) +
  scale_color_manual(values = c("#424242","#a8404c")) +
  theme(legend.position = "bottom")

differentiation_heatmap <- pheatmap(rld_df, 
                                    show_rownames = FALSE, 
                                    scale = "row", 
                                    color = col_pal,
                                    treeheight_row = 0, 
                                    cutree_cols = 6, 
                                    silent = TRUE)


text <- ggparagraph(text = "The transcriptomes of lncMYH7b KD cells diverge from control cells late in the differentiation timecourse. A. PCA performed on normalized gene counts at each timepoint and condition for the subset of highly variable genes (>= 90th-percentile variance). B. Heatmap of normalized gene counts for each timepoint and condition for the highly variable genes. At day 2, 4, and 6, lncMYH7b KD cells cluster with control cells. Toward the end of the differentiation (days 10 and 12) KD cells cluster irregularly. KD cells on day 10 form an outgroup of the CTLd10/12 cluster. KD cells on D12 form an outgroup of the CTLd6/10/12 cluster.", size = 11, color = "black", family = "sans")
fig1 <- ggarrange(ggarrange(differentiation_pca, differentiation_heatmap[[4]],
                            labels = c("A", "B"),
                            ncol = 2, nrow = 1), text,
                  ncol = 1, nrow = 2,
                  heights = c(1, 0.3)) +
  theme(plot.margin = margin(1,1,1,1, "cm")) 
fig1
ggsave("figures/transcriptome_concordance.pdf")
```

## How many genes does lncMYH7b KD effect over the course of differentiation?

To answer this question, we will look at how many genes are changing
over the course of differentiation between the KD cells and the controls
at each timepoint. Given that we have sufficient replicates for each
timepoint (4 replicates), we can make this comparison with good depth.

```{r run-deseq, include=FALSE}
# Since we don't have a zero timepoint... this is not ideal. 
# But since it is the case that the 2 day timepoint is quite similar
# between the two conditions, we can compare back to this timepoint
# and treat it as a zero timepoint. May need to make a supp. fig. here
# to back this up.
# What we will do here is look at how many genes are changing at each timepoint
# compared to the two day timepoint.
# We will run this seperately for the KD cells and the CTL cells.
# And we have quite enough replicates to get good depth here. 

# CTL cells
ctl_samples <- samples %>% filter(condition == "CTL")
ctl_counts <- salmon_counts[,ctl_samples$sample_id]
dds <- DESeqDataSetFromMatrix(countData = ctl_counts,
                              colData = ctl_samples,
                              design = ~ timepoint,
                              rowData = genes)
dds <- DESeq(dds)
res_to_complile <- resultsNames(dds)[grepl("timepoint", resultsNames(dds))]
ctl_res_df <- lapply(res_to_complile, function(x) results(dds, name = x) %>%
                       as.data.frame() %>%
                       rownames_to_column("gene_id") %>%
                       left_join(g2s) %>%
                       mutate(result_name = x,
                              condition = "ctl",
                              lfc_type = "lfc")) %>%
  bind_rows()
ctl_res_shrnk <- lapply(res_to_complile, function(x) lfcShrink(dds, coef = x, type = "ashr") %>%
                       as.data.frame() %>%
                       rownames_to_column("gene_id") %>%
                       left_join(g2s) %>%
                       mutate(result_name = x,
                              condition = "ctl",
                              lfc_type = "lfc_shrnk")) %>%
  bind_rows()

# KD cells -- for completeness *not actually going to use this here*
kd_samples <- samples %>% filter(condition == "myh7b_KD")
kd_counts <- salmon_counts[,kd_samples$sample_id]
dds <- DESeqDataSetFromMatrix(countData = kd_counts,
                              colData = kd_samples,
                              design = ~ timepoint,
                              rowData = genes)
dds <- DESeq(dds)
res_to_complile <- resultsNames(dds)[grepl("timepoint", resultsNames(dds))]
kd_res_df <- lapply(res_to_complile, function(x) results(dds, name = x) %>%
                      as.data.frame() %>%
                      rownames_to_column("gene_id") %>%
                      left_join(g2s) %>%
                      mutate(result_name = x,
                             condition = "kd")) %>%
  bind_rows()
kd_res_shrnk <- lapply(res_to_complile, function(x) lfcShrink(dds, coef = x, type = "ashr") %>%
                       as.data.frame() %>%
                       rownames_to_column("gene_id") %>%
                       left_join(g2s) %>%
                       mutate(result_name = x,
                              condition = "ctl",
                              lfc_type = "lfc_shrnk")) %>%
  bind_rows()
vs_d2_res_df <- bind_rows(ctl_res_df, ctl_res_shrnk, kd_res_df, kd_res_shrnk)

# Tidy up the result name into a timepoint
vs_d2_res_df <- vs_d2_res_df %>%
  mutate(comparison = gsub("timepoint_", "", result_name),
         comparison = factor(comparison, levels = c("d4_vs_d2", "d6_vs_d2", "d10_vs_d2", "d12_vs_d2")))
write_csv(vs_d2_res_df, "results/vs_d2_deseq_results.csv")


# Okay, the second part of this that we need is the individual timepoint comparisons
# between the KD cells and the CTLs
deseq_timepoint  <- function(tp, samples, counts) {
  
  tp_samples <- samples %>% filter(timepoint == tp)
  tp_counts <- counts[,tp_samples$sample_id]
  dds <- DESeqDataSetFromMatrix(countData = tp_counts,
                                colData = tp_samples,
                                design = ~ condition,
                                rowData = genes)
  dds <- DESeq(dds)
  res <- results(dds, name = "condition_myh7b_KD_vs_CTL") %>%
    as.data.frame() %>%
    rownames_to_column("gene_id") %>%
    left_join(g2s) %>%
    mutate(timepoint = tp,
           lfc_type = "lfc")
  res_shrnk <- lfcShrink(dds, coef = "condition_myh7b_KD_vs_CTL", type = "ashr") %>%
        as.data.frame() %>%
    rownames_to_column("gene_id") %>%
    left_join(g2s) %>%
    mutate(timepoint = tp,
           lfc_type = "lfc_shrnk") 
  return(bind_rows(res, res_shrnk))
}

timepoints <- as.character(unique(samples$timepoint))
tp_results <- lapply(timepoints, function(x) deseq_timepoint(x, samples, salmon_counts)) %>%
  bind_rows()
write_csv(tp_results, "results/timepoint_kd_vs_ctl_deseq_results.csv")
```

```{r num-deg, echo = FALSE, message = FALSE, warning=FALSE}
# Let's make a plot of how many up and downregulated genes there are at each timepoint
# for KD vs CTL
tp_sig <- tp_results %>%
  filter(lfc_type == "lfc",
    padj < 0.01, 
         abs(log2FoldChange) > 1) %>%
  mutate(deg = ifelse(log2FoldChange > 0, "up", "down"))
tp_sig_summary <- tp_sig %>%
  group_by(timepoint, deg) %>%
  summarize(ndeg = n())
tp_sig_summary$timepoint <- factor(tp_sig_summary$timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))


tp_sig_summary$ndeg[tp_sig_summary$deg == "down"] <- tp_sig_summary$ndeg[tp_sig_summary$deg == "down"] * -1
ndeg_barplot <- ggplot(tp_sig_summary, aes(x = timepoint, y = ndeg, fill = deg)) +
  geom_hline(yintercept = 0) +
  geom_bar(stat = "identity") +
  theme_paperwhite() +
  scale_fill_manual(values = c("#424242","#a8404c")) +
  geom_text(data = tp_sig_summary %>% filter(deg == "up"),stat='identity', aes(label=ndeg), vjust=-1) +
    geom_text(data = tp_sig_summary %>% filter(deg == "down"),stat='identity', aes(label=ndeg*-1), vjust=1.5) +
  xlab("") +
  ylab("Number of DEG")

# Let's make a heatmap of these genes
all_sig_tp_genes <- unique(tp_sig$gene_id)

lfc_shrnk_tp_sig <- tp_results %>%
  filter(lfc_type == "lfc_shrnk",
         gene_id %in% all_sig_tp_genes) %>%
  dplyr::select(gene_id, log2FoldChange, timepoint) %>%
  pivot_wider(names_from = timepoint, values_from = log2FoldChange, values_fill = 0) %>%
  column_to_rownames("gene_id") %>%
  as.matrix()

tp_lfc_heatmap <- pheatmap(lfc_shrnk_tp_sig, cluster_cols = FALSE, color = col_pal, show_rownames = FALSE,
         breaks = seq(-5,5,length.out = 100), silent = TRUE)

nrow(lfc_shrnk_tp_sig)

text <- ggparagraph(text = "There are very few differentially expressed genes between KD cells and controls until day 10 of differentiation. A. Number of differentially expressed genes between lncMYH7b cells and controls within each timepoint (padj < 0.01; |log2FC| > 1). B. Heatmap of shrunken log2FoldChanges for all genes differentially expressed at any one timepoint(4,452 genes).", size = 11, color = "black", family = "sans")
fig1 <- ggarrange(ggarrange(ndeg_barplot, tp_lfc_heatmap[[4]],
                            labels = c("A", "B"),
                            ncol = 2, nrow = 1, widths = c(0.7, 0.9)), text,
                  ncol = 1, nrow = 2,
                  heights = c(1, 0.3)) +
  theme(plot.margin = margin(1,1,1,1, "cm")) 
fig1
```

# How to the hallmark marker genes for cardiomyocyte differentiation compare in the KD cells and controls?

Here we will just plot the 24 marker genes, and their relative
expression compared to control

```{r marker-genes, echo = FALSE, message = FALSE, warning=FALSE}}
marker_genes <- c("Pou5f1", "Nanog", "Sox2",
                  "Tbxt", "Mesp1", "Isl1",
                  "Kdr", "Mef2c",
                  "Gata4", "Nkx2-5", "Tbx3", "Tbx5",
                  "Tbx20", "Irx4",
                  "Nr2f2",
                  "Tnnt2", 
                  "Myh7", 
                  "Myl2",
                  "Myl7",
                  "Myl3", "Myl4",
                  "Shox", "Hcn3", "Hcn4",
                  "Cacna1d", "Ryr2", "ATP2A2") %>%
  toupper()
# TODO: double check that these are the right genes. Particularly Serca == ATP2A2
marker_g2s <- g2s %>% filter(gene_name %in% marker_genes) %>%
  # There are two matches for Shox -- also need to make sure this is the right one.
  filter(gene_id != "ENSG00000185960.14_PAR_Y")

# We can either plot the fold-changes or the tpm here. 
# I suppose let's start with TPM.
salmon_tpm <- read_csv("../../results/salmon/salmon_merged_gene_tpm.csv") %>%
  pivot_longer(2:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  left_join(g2s) %>% 
  filter(gene_id %in% marker_g2s$gene_id) %>%
  left_join(samples)
salmon_tpm$timepoint <- factor(salmon_tpm$timepoint, levels = c("d2", "d4", 
                                                                "d6", "d10",
                                                                "d12"))

# Merge in DE genes
salmon_tpm <- salmon_tpm %>% 
  unite(gene_tp, gene_id, timepoint, remove = FALSE)
tp_sig <- tp_sig %>% unite(gene_tp, gene_id, timepoint, remove = FALSE)

salmon_tpm <- salmon_tpm %>% left_join(tp_sig %>% dplyr::select(gene_tp, deg))
salmon_tpm$deg[!is.na(salmon_tpm$deg)] <- "sig"

salmon_tpm_summary <- salmon_tpm %>% 
  group_by(gene_name, condition, timepoint, deg) %>%
  summarize(mean_tpm = mean(tpm),
            sd_tpm = sd(tpm))
max_tpm_per_gene  <- salmon_tpm_summary %>%
  group_by(gene_name) %>%
  summarize(max_tpm = max(mean_tpm))
salmon_tpm_summary <- salmon_tpm_summary %>%
  left_join(max_tpm_per_gene)

# Pluripotency markers - oct4, sox2, nanog
#  cardiac mesodermal commitment  -- TBXT MESP1
# cardiac transcriptional factors -- Isl1, Kdr, Mef2C, NKx2.5, Gata4, Tbx3/5/20 
#  ventricular and atrial specific transcriptional factor -- Irf4 Nr2f2
# Cardiac structural and sarcormeric proteins -- Tnni2, Mlc2a/v, Myh7, Myl3/4
salmon_tpm_summary$gene_name <- factor(salmon_tpm_summary$gene_name, levels = marker_genes)
salmon_tpm$gene_name <- factor(salmon_tpm$gene_name, levels = marker_genes)
ggplot(salmon_tpm, aes(x = timepoint, y = tpm, color = condition, group = condition)) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = "d2") +
  # geom_errorbar(aes(ymin = mean_tpm - sd_tpm, ymax = mean_tpm + sd_tpm)) +
  geom_smooth(se = T, method = 'loess', formula = 'y ~ x', alpha = 0.1) + 
  facet_wrap(~gene_name, ncol = 6, scales = "free_y") + 
  scale_color_manual(values = c("#424242","#a8404c")) +
  theme_paperwhite(grid = TRUE) +
  geom_text(data = salmon_tpm_summary %>% 
              filter(deg == "sig"), aes(label = "*", y = max_tpm), stat = "identity",
            size = 7, show.legend = FALSE) +
  scale_y_continuous(expand = c(0.1, 2))
# ATP2A2 should have *
```
