---
title: "01_lncMYH7b_KD_vs_control_timecourse"
author: "Michael Smallegan"
date: "9/23/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(DESeq2)
library(rtracklayer)
library(janitor)
library(ggrepel)
library(pheatmap)
library(ggpubr)
source("../../util/_plot_theme.R")
```

```{r data-import, include=FALSE}
samples <- read_csv("../diff_samplesheet.csv") %>%
  clean_names() %>%
  # Make some factors out of the relevant variables.
  mutate(condition = factor(condition, levels = c("CTL", "myh7b_KD")),
         timepoint = factor(timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))) %>% 
  # Put the columns in order
  arrange(condition, timepoint, bio_replicate) %>%
  # Add a more meaningful name for plots
  unite(sample_name, timepoint, condition, bio_replicate, remove = FALSE) %>%
  mutate(sample_name = gsub("myh7b_", "", sample_name))

# Grab the counts
salmon_counts <- read_csv("../../results/salmon/salmon_merged_gene_counts.csv") %>%
  column_to_rownames("gene_id") %>%
  as.matrix()

# Filter genes with less than 10 counts across the whole dataset
salmon_counts <- salmon_counts[rowSums(salmon_counts) > 10,]

# Get the annotations
# gtf <- rtracklayer::import("../../util/gencode.v33.annotation.gtf")
# genes <- gtf[gtf$type == "gene"]
# rtracklayer::export(genes, "results/gencode_v33_genes.gtf")
genes <- rtracklayer::import("results/gencode_v33_genes.gtf")
# g2s <- genes %>% 
#   as.data.frame() %>%
#   dplyr::select(gene_id, gene_name)
# write_csv(g2s, "results/gene2symbol.csv")
g2s <- read_csv("results/gene2symbol.csv")

# Line it all up
names(genes) <- genes$gene_id
genes <- genes[rownames(salmon_counts)]
stopifnot(all(names(genes) == rownames(salmon_counts)))

samples <- samples %>% as.data.frame()
rownames(samples) <- samples$sample_id
salmon_counts <- salmon_counts[,rownames(samples)]
stopifnot(all(rownames(samples) == colnames(salmon_counts)))

# Change count matrix to integer mode
mode(salmon_counts) <- "integer"
```

# lncMYH7b KD during cardiomyocyte differentiation

We observed that lncMYH7b is required for iPSCs to properly
differentiate into cardiomyocytes. Comparing the transcriptome over the
course of differentiation from lncMYH7b KD iPSCs to control iPSCs can
answer a few key questions:

1.  Globally, how well do the transcriptomes of lncMYH7b KD cells and
control cells concord over the differentiation timecourse?

1.  At what stage of differentiation do lncMYH7b KD cells start to
diverge from controls?

2.  To what extent do the genes involved in lineage commitment
diverge from controls?

3.  Are lncMYH7b KD cells "stuck" in the middle of lineage
commitment or do they start to look like an entirely different
cell type?

2.  Are there steps early in the differentiation process that are
misregulated by lncMYH7b depletion?

1.  Since other results indicate that the WNT signalling pathway may
be implicated:

1.  Is the WNT signalling pathway affected?

2.  Do gene expression changes give us an insight into how the
pathway is affected?

3.  In the final stages of differentiation, are there genes or pathways
that contribute to explaining the observed phenotype?

## Transcriptome concordance over differentiation

Here we will take a look at the mean expression across replicates for
each timepoint and condition in a PCA plot and heatmap to determine to
what extent the lncMYH7b cells are following the same differentiation
pathway and how closely their transcriptomes end up resembling healthy
cardiomyocytes.

```{r transcriptome-concordance, include=FALSE}
# Here we will use the regularized log normalized counts from DESeq2
# For that we need to make a DESeqDataSet object.
dds <- DESeqDataSetFromMatrix(countData = salmon_counts,
                              colData = samples,
                              design = ~ timepoint + condition + timepoint*condition,
                              rowData = genes)
# * commented out for speed in knitting * #
# rld <- rlog(dds, blind = FALSE)
# write_rds(rld, "results/rld_counts.rds")
rld <- read_rds("results/rld_counts.rds")
rld_matrix <- assay(rld)

# PCA plot with the mean normalized count across each sample.
# For this we will take just the genes which have high variance, 
# which is more informative for the trajectory of the transcriptomes.
# Calculate the variance across each gene. 
gene_var <- rowVars(rld_matrix)
# Take genes with variance in the 75th-percentile and up 
rld_var <- rld_matrix[gene_var >= quantile(gene_var, 0.9),]

# Merge replicates based on the mean normalized counts across the replicates.
rld_df <- rld_var %>% 
  as.data.frame() %>% 
  rownames_to_column("gene_id") %>%
  pivot_longer(2:ncol(.), names_to = "sample_id", values_to = "rld") %>%
  left_join(samples) %>%
  unite(tcondition, condition, timepoint) %>%
  group_by(gene_id, tcondition) %>%
  summarize(mean_rld = mean(rld)) %>%
  pivot_wider(names_from = tcondition, values_from = mean_rld) %>%
  column_to_rownames("gene_id") %>%
  as.matrix()

# PCA
pca <- prcomp(rld_df, rank = 2)

# Get the proportion of variance explained by each principle component.
pca_summary <- summary(pca)$importance
prop_variance <- round(pca_summary["Proportion of Variance", 1:2] * 100)

# Collect into a data.frame for plotting
pca_df <- pca$rotation %>% 
  as.data.frame() %>%
  rownames_to_column("tcondition") %>%
  mutate(condition = ifelse(grepl("CTL", tcondition), "ctl", "kd"),
         timepoint = gsub("CTL_|myh7b_KD_", "", tcondition))
pca_df$timepoint <- factor(pca_df$timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))
pca_df$condition <- factor(pca_df$condition, levels = c("ctl", "kd"))

## Make a figure
differentiation_pca <- ggplot(pca_df, aes(x = PC1, y = PC2, color = condition, label = timepoint)) +
  geom_point(size = 3) +
  geom_text_repel() +
  theme_paperwhite() +
  xlab(paste0("PC1: ", prop_variance[[1]], "% variance")) +
  ylab(paste0("PC2: ", prop_variance[[2]], "% variance")) +
  scale_color_manual(values = c("#424242","#a8404c")) +
  theme(legend.position = "bottom")

differentiation_heatmap <- pheatmap(rld_df, 
                                    show_rownames = FALSE, 
                                    scale = "row", 
                                    color = col_pal,
                                    treeheight_row = 0, 
                                    cutree_cols = 6, 
                                    silent = TRUE)


text <- ggparagraph(text = "The transcriptomes of lncMYH7b KD cells diverge from control cells late in the differentiation timecourse. A. PCA performed on normalized gene counts at each timepoint and condition for the subset of highly variable genes (>= 90th-percentile variance). B. Heatmap of normalized gene counts in timepoint and condition for the highly variable genes. At day 2, 4, and 6, lncMYH7b KD cells cluster with control cells. Toward the end of the differentiation (days 10 and 12) KD cells cluster irregularly. KD cells on day 10 form an outgroup of the CTLd10/12 cluster. KD cells on D12 form an outgroup of the CTLd6/10/12 cluster.", size = 11, color = "black", family = "sans")
fig1 <- ggarrange(ggarrange(differentiation_pca, differentiation_heatmap[[4]],
                            labels = c("A", "B"),
                            ncol = 2, nrow = 1), text,
                  ncol = 1, nrow = 2,
                  heights = c(1, 0.3)) +
  theme(plot.margin = margin(1,1,1,1, "cm")) 
fig1
ggsave("figures/transcriptome_concordance.pdf")
```

## How many genes does lncMYH7b KD effect over the course of differentiation?

To answer this question, we will look at how many genes are changing
over the course of differentiation in the control cells and percentage
of those that are misregulated by lncMYH7b.

```{r}
# Since we don't have a zero timepoint... this is not ideal. 
# But since it is the case that the 2 day timepoint is quite similar
# between the two conditions, we can compare back to this timepoint
# and treat it as a zero timepoint. May need to make a supp. fig. here
# to back this up.
# What we will do here is look at how many genes are changing at each timepoint
# compared to the two day timepoint.
# We will run this seperately for the KD cells and the CTL cells.
# And we have quite enough replicates to get good depth here. 

# CTL cells
ctl_samples <- samples %>% filter(condition == "CTL")
ctl_counts <- salmon_counts[,ctl_samples$sample_id]
dds <- DESeqDataSetFromMatrix(countData = ctl_counts,
                              colData = ctl_samples,
                              design = ~ timepoint,
                              rowData = genes)
dds <- DESeq(dds)
res_to_complile <- resultsNames(dds)[grepl("timepoint", resultsNames(dds))]
ctl_res_df <- lapply(res_to_complile, function(x) results(dds, name = x) %>%
                       as.data.frame() %>%
                       rownames_to_column("gene_id") %>%
                       left_join(g2s) %>%
                       mutate(result_name = x,
                              condition = "ctl")) %>%
  bind_rows()

# KD cells -- for completeness *not actually going to use this here*
samples$condition
kd_samples <- samples %>% filter(condition == "myh7b_KD")
kd_counts <- salmon_counts[,kd_samples$sample_id]
dds <- DESeqDataSetFromMatrix(countData = kd_counts,
                              colData = kd_samples,
                              design = ~ timepoint,
                              rowData = genes)
dds <- DESeq(dds)
res_to_complile <- resultsNames(dds)[grepl("timepoint", resultsNames(dds))]
kd_res_df <- lapply(res_to_complile, function(x) results(dds, name = x) %>%
                      as.data.frame() %>%
                      rownames_to_column("gene_id") %>%
                      left_join(g2s) %>%
                      mutate(result_name = x,
                             condition = "kd")) %>%
  bind_rows()

vs_d2_res_df <- bind_rows(ctl_res_df, kd_res_df)

# Tidy up the result name into a timepoint
vs_d2_res_df <- vs_d2_res_df %>%
  mutate(comparison = gsub("timepoint_", "", result_name),
         comparison = factor(comparison, levels = c("d4_vs_d2", "d6_vs_d2", "d10_vs_d2", "d12_vs_d2")))
write_csv(vs_d2_res_df, "results/vs_d2_deseq_results.csv")


# Okay, the second part of this that we need is the individual timepoint comparisons
# between the KD cells and the CTLs
deseq_timepoint  <- function(tp, samples, counts) {
  
  tp_samples <- samples %>% filter(timepoint == tp)
  tp_counts <- counts[,tp_samples$sample_id]
  dds <- DESeqDataSetFromMatrix(countData = tp_counts,
                                colData = tp_samples,
                                design = ~ condition,
                                rowData = genes)
  dds <- DESeq(dds)
  res <- results(dds, name = "condition_myh7b_KD_vs_CTL") %>%
    as.data.frame() %>%
    rownames_to_column("gene_id") %>%
    left_join(g2s) %>%
    mutate(timepoint = tp)
  return(res)
}

timepoints <- as.character(unique(samples$timepoint))
tp_results <- lapply(timepoints, function(x) deseq_timepoint(x, samples, salmon_counts)) %>%
  bind_rows()
write_csv(tp_results, "results/timepoint_kd_vs_ctl_deseq_results.csv")


tp_results <- tp_results %>%
  mutate(timepoint = factor(timepoint, levels = c("d2", "d4", "d6", "d10", "d12")))

sig_tp_results <- tp_results %>%
  filter(padj < 0.01, abs(log2FoldChange) > 1)
ggplot(sig_tp_results, aes(x = timepoint)) +
  geom_bar() +
geom_text(stat='count', aes(label=..count..), vjust=-1)


ctl_sig <- vs_d2_res_df %>%
  filter(condition == "ctl",
         padj < 0.01,
         abs(log2FoldChange) > 1)

# Cummulatively how many genes change through differentiation?
d4_genes <- ctl_sig %>% filter(comparison == "d4_vs_d2") %>%
  pull(gene_id)
kd_d4_genes <- kd_sig %>% filter(comparison == "d4_vs_d2") %>%
  pull(gene_id)


d6_genes <- ctl_sig %>% filter(comparison == "d6_vs_d2") %>%
  pull(gene_id)
kd_d6_genes <- kd_sig %>% filter(comparison == "d6_vs_d2") %>%
  pull(gene_id)
cum_d6_kd <- unique(c(kd_d4_genes, kd_d6_genes))

length(d4_genes)
d6_cum <- unique(c(d4_genes, d6_genes))


length(d6_genes)
length(d6_cum)
length(which(d4_genes %in% d6_genes))
d10_genes <- ctl_sig %>% filter(comparison == "d10_vs_d2") %>%
  pull(gene_id)
d10_cum <- unique(c(d4_genes, d6_genes, d10_genes))
kd_d10_genes <- kd_sig %>% filter(comparison == "d10_vs_d2") %>%
  pull(gene_id)
cum_d10_kd <- unique(c(kd_d4_genes, kd_d6_genes, kd_d10_genes))
length(which(cum_d10_kd %in% d10_cum))

length(d10_cum)
d12_genes <- ctl_sig %>% filter(comparison == "d12_vs_d2") %>%
  pull(gene_id)
d12_cum <- unique(c(d4_genes, d6_genes, d10_genes, d12_genes))
kd_d12_genes <- kd_sig %>% filter(comparison == "d12_vs_d2") %>%
  pull(gene_id)
cum_d12_kd <- unique(c(kd_d4_genes, kd_d6_genes, kd_d10_genes, kd_d12_genes))
length(which(cum_d12_kd %in% d12_cum))
length(d12_cum)

ctl_cum_deg <- data.frame(timepoint = c("d2", "d4", "d6", "d10", "d12"),
                          cummulative_deg = c(0, length(d4_genes),
                                        length(d6_cum),
                                        length(d10_cum),
                                        length(d12_cum)),
                          cummulative_deg_kd = c(0,
                                                 length(which(kd_d4_genes %in% d4_genes)),
                                                 length(which(cum_d6_kd %in% d6_cum)),
                                                 length(which(cum_d10_kd %in% d10_cum)),
                                                 length(which(cum_d12_kd %in% d12_cum)))) %>%
  mutate(percent_deg_in_kd = round((cummulative_deg_kd / cummulative_deg)*100),
         percent_deg_in_kd = paste0(percent_deg_in_kd, "%")) 
ctl_cum_deg$percent_deg_in_kd[1] <- ""
ctl_cum_deg$timepoint <- factor(ctl_cum_deg$timepoint, levels = c("d2", "d4", "d6", "d10", "d12"))

ggplot(ctl_cum_deg, aes(x = timepoint, y = cummulative_deg, group = 1, label = percent_deg_in_kd)) + 
  geom_line()+
   geom_area(fill = "#424242") +
   geom_text(stat='identity', vjust=-1) +
  geom_area(aes(y = cummulative_deg_kd), fill= "#a8404c", alpha = 0.8) +
  theme_paperwhite()

ggplot(ctl_sig, aes(x = comparison)) +
  geom_bar()

kd_sig <- vs_d2_res_df %>%
  filter(condition == "kd",
         padj < 0.01,
         abs(log2FoldChange) > 1)


kd_sig_d4 <- kd_sig %>%
  filter(comparison == "d4_vs_d2")
ctl_sig_d4 <- ctl_sig %>%
  filter(comparison == "d4_vs_d2")
nrow(kd_sig_d4)
nrow(ctl_sig_d4)
length(which(kd_sig_d4$gene_id %in% ctl_sig_d4$gene_id)) / nrow(ctl_sig_d4)
length(ctl_sig_d4$gene_id %in% kd_sig_d4$gene_id) / nrow(kd_sig_d4)
ggplot(kd_sig, aes(x = comparison)) +
  geom_bar()

```


## Is the WNT signalling pathway affected by lncMYH7b KD?

### In what manner is the WNT signalling pathway misregulated?

## Which genes/pathways contribute to the observed phenotype?

```{r}
# Let's get the genes associated with WNT signalling

# WNT signalling pathway: GO:0016055 (455 genes)
# positive regulation of Wnt signaling pathway by Wnt receptor internalization: GO:0038013
# regulation of Wnt signaling pathway by Wnt protein secretion: GO:0061359
# positive regulation of Wnt signaling pathway: GO:0030177
# positive regulation of Wnt signaling pathway by BMP signaling pathway: GO:0060804
# Wnt signaling pathway involved in heart development: GO:0003306
# regulation of Wnt signaling pathway: GO:0030111
# Wnt-activated receptor activity: GO:0042813
# negative regulation of Wnt signaling pathway: GO:0030178
# canonical Wnt signaling pathway: GO:0060070
# negative regulation of Wnt signaling pathway by Wnt receptor internalization: GO:0038012
library(biomaRt)
ensembl = useMart("ensembl",dataset="hsapiens_gene_ensembl")
gene.data <- getBM(attributes=c('hgnc_symbol', 'ensembl_gene_id', 'go_id'),
                   filters = 'go', values = 'GO:0007507', mart = ensembl)

```

Purpose: look at which genes are changing differently in time in the
lncMYH7b KD condition vs control in iPSC to cardiomyocyte
differentiation. I think there are three questions we can answer at this
point: 1. During the course of differentiation There are actually two
questions to ask here: 1. Which genes change early and can influence the
lineage commitment pathway during differentiation? 2. Which genes are
most robustly changing over time -- in other words, which are the most
different over differentiation from the control cells?

For question one, the way you were approaching it will work because we
have enough replicates to compare directly KD to CTL at each timepoint.
For question two, we can use a timecourse approach to look at which
genes are responding most dramatically. We can then cluster these and
see if there are some pathways enriched that can explain how these
lncMYH7b KD cells differ from normal cardiomyocytes.

This analysis will give us genes that are responding differently over
time to the differentiation timecourse in the KD condition vs control.
In this list of genes, those with small p-values are those which at one
or more time points after time 2 days showed a KD-specific effect.

```{r deg-timecourse}
# Run DESeq2 to get genes DE in time in the KD compared to control
dds <- DESeqDataSetFromMatrix(countData = salmon_counts,
                              colData = samples,
                              design = ~ timepoint + condition + timepoint*condition,
                              rowData = genes)
dds <- DESeq(dds, test = "LRT", reduced = ~ timepoint + condition)

# Retrieve the LRT results.
lrt_res <- results(dds, tidy = TRUE) %>%
  dplyr::rename(gene_id = row) %>%
  left_join(g2s) %>%
  dplyr::select(gene_id, gene_name, everything())
write_csv(lrt_res, "results/lrt_res.csv")

# Let's make some cutoffs. These are flexible obviously.
lrt_sig <- lrt_res %>% 
  filter(padj < 0.01, abs(log2FoldChange) > 1)
write_csv(lrt_sig, "results/lrt_res_sig.csv")
```

Let's make a quick heatmap. We can plot the genes that changed
differently in time in the KD with padj \< 0.01 and abs(log2FC) \> 1.
Which is 1614 genes. This isn't the best heatmap for visualizing what is
happening because it is still pretty raw data, however, it does give a
good overview of exactly how much data went into this and might be good
for the supp. We'll use here the regularized log count from DESeq2,
shown z-scaled across each gene.

```{r deg-tc-genes, message=FALSE}


# Make sure it is in the proper order and replace the sample id with the sample name
sig_rld_matrix <- sig_rld_matrix[,samples$sample_id]
colnames(sig_rld_matrix) <- samples$sample_name

ht1 <- pheatmap::pheatmap(sig_rld_matrix, 
                          scale = "row", 
                          show_rownames = FALSE, 
                          cluster_cols = FALSE,
                          color = col_pal, 
                          main = "Differentially expressed genes in time in lncMY7b KD")
ht1
pdf("figures/deg_timecourse_padj0.01_heatmap.pdf"); ht1; dev.off()
```

Let's make a PCA plot with all the samples. This first one will contain
replicates, but what we can do is take the mean normalized count across
each timepoint and condition and plot that, so that we can show the
trajectory through the space and show how they diverge.

```{r}
DESeq2::plotPCA(rld, intgroup = "condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("Differentiation PCA")





# PCA annotated with all the replicates unmerged
rld_matrix <- assay(rld)
d2_samples <- samples %>% filter(timepoint == "d6")
d2_matrix <- rld_matrix[,d2_samples$sample_id]
pca <- prcomp(d2_matrix)
pca_df <- pca$rotation %>% as.data.frame() %>%
  rownames_to_column("sample_id") %>%
  left_join(samples)
ggplot(pca_df, aes(x = PC1, y = PC2, color = condition, label = sample_id)) +
  geom_point() +
  geom_text_repel() +
  theme_paperwhite() +
  scale_color_manual(values = c("#424242","#a8404c"))
```

```{r}
# clustering of samples variable genes
rld_scaled <- t(scale(t(rld_df)))
pheatmap(rld_scaled, show_rownames = FALSE, scale = "none", color = col_pal)
```

We only get one p-value for each gene, but we do get log2FoldChanges for
each timepoint. One has to think a little bit about how to interpret
these because they are the fold-change for KD vs CTL at each timepoint
with the overall time trend and the differences at time zero (or day 2
in this case) subtracted out. Might not be the best for showing in
publication plots, but I find it's a bit helpful.

```{r extract-lfc-results}
# ** This all takes a while to run, so I've commented it out for the purposes of knitting ** #

# # Extract list of LFCs for each timepoint vs 2d and CTL.
# # Let's also get the shrunken LFC here, since that is a nice way of visualizing those
# # genes for which we have high confidence that they're changing.
# res2compile <- resultsNames(dds)[grep("conditionmyh7b_KD", resultsNames(dds))]
# res_list <- lapply(res2compile, function(x) results(dds, name = x))
# shrnk_res_list <- lapply(res2compile, function(x) lfcShrink(dds = dds, coef = x, type="apeglm"))
# # Then let's bind these all up in one dataframe to make life easier.
# for(i in 1:length(res_list)) {
#   tmp_df <- res_list[[i]] %>% 
#     as.data.frame() %>%
#     rownames_to_column("gene_id") %>%
#     left_join(g2s) %>%
#     mutate(lfc_type = "lfc",
#            result_name = res2compile[[i]])
#   tmp_shrnk_df <- shrnk_res_list[[i]] %>% 
#     as.data.frame() %>%
#     rownames_to_column("gene_id") %>%
#     left_join(g2s) %>%
#     mutate(lfc_type = "lfc_shrnk",
#            result_name = res2compile[[i]])
#   res_list[[i]] <- bind_rows(tmp_df, tmp_shrnk_df)
# }
# res_df <- bind_rows(res_list)
# write_csv(res_df, "results/lrt_timepoint_lfc.csv")
res_df <- read_csv("results/lrt_timepoint_lfc.csv")
```

Let's make some plots of this data.

```{r}
# Let's clean up the comparison column a bit to get the timepoints
res_df <- res_df %>%
  mutate(timepoint = gsub("timepoint|.conditionmyh7b_KD", "", result_name),
         timepoint = factor(timepoint, levels = c("d4", "d6", "d10", "d12")))


# First let's just look at how many genes hit a l2fc threshold in each 
# timepoint
shrnk_res_df <- res_df %>% filter(lfc_type == "lfc_shrnk")
shrnk_res_sig <- shrnk_res_df %>% filter(padj < 0.01, abs(log2FoldChange) > 1) %>%
  mutate(deg = ifelse(log2FoldChange > 0, "up", "down"))

res_sig_summary <- shrnk_res_sig %>%
  group_by(timepoint, deg) %>%
  summarize(ngenes = n())
ggplot(res_sig_summary, aes(x = deg, y = ngenes)) +
  geom_bar(stat = "identity") +
  facet_grid(~timepoint) + 
  geom_text(stat='identity', aes(label=ngenes), vjust=-1) +
  theme_paperwhite() + 
  ggtitle("Number of DEG KD vs CTL in time")
```

Great, let's make some plots of those genes that are changing at these
early timepoints since it is few enough to visualize.

```{r}
tpm <- read_csv("../../results/salmon/salmon_merged_gene_tpm.csv")

d4_sig <- shrnk_res_sig %>%
  filter(timepoint == "d4")
d6_sig <- shrnk_res_sig %>%
  filter(timepoint == "d6")
# Weirdly it is an entirely different set of genes.
# There is a lot happening in these cells in these few days.
table(d4_sig$gene_id %in% d6_sig$gene_id)

d4_sig_tpm <- tpm %>% filter(gene_id %in% d4_sig$gene_id) %>%
  pivot_longer(2:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  left_join(samples) %>%
  left_join(g2s)
ggplot(d4_sig_tpm, aes(x = timepoint, y = tpm, color = condition, group = condition)) +
  geom_point() +
  geom_smooth(se = FALSE) +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) +
  facet_wrap(~gene_name, scales = "free_y") +
  theme_paperwhite() +
  scale_color_manual(values = c("#424242","#a8404c"))

```

Can't help myself with a little bit of looking at what's going on here,
but I'll leave that up to you. -MS

```{r, include = FALSE}
# Output for DAVID
lrt_genes <- lrt_sig %>%
  dplyr::select(gene_id) 
lrt_genes$gene_id <- sapply(lrt_genes$gene_id, function(x) unlist(strsplit(x, ".", fixed = T))[[1]])
write.table(lrt_genes, "results/lrt_sig_gene_ids.txt", quote = FALSE,
            col.names = FALSE, row.names = FALSE)

bg_genes <- rownames(salmon_counts)
bg_genes <- sapply(bg_genes, function(x) unlist(strsplit(x, ".", fixed = T))[[1]])
write.table(data.frame(gene_id = bg_genes), "results/background_gene_ids.txt",
            quote = FALSE,
            col.names = FALSE, row.names = FALSE)
```

```{r}

?pheatmap
rld_matrix[[1]]
fiss <- plotCounts(dds, which.min(res$padj), 
                   intgroup = c("timepoint","condition"), returnData = TRUE)
fiss$timepoint
ggplot(fiss,
       aes(x = timepoint, y = count, color = condition, group = condition)) + 
  geom_point() + stat_summary(fun.y=mean, geom="line") +
  scale_y_log10()


```
