---
title: "Differential gene analysis, lncMYH7b KD during differentiation"
output: html_document
---

```{r setup, message=FALSE}
options(stringsAsFactors = FALSE)
library(data.table)
setDTthreads(threads = 12)

library(rcartocolor) 
library(genefilter)
library(tximport)
library(tidyverse)
library(DESeq2)
library(pheatmap)
library(grDevices)
library(ggrepel)
library(venneuler)
library(ggforce)
library(ggpubr)

source("../util/_util.R")
source("../util/_plotting.R")
```


```{r import_counts, message=FALSE, warning=FALSE, results='hide'}
# Read in gencode annotations
gtf <- rtracklayer::import("../util/gencode.v33.annotation.gtf")

# Renaming for compatibility with tximport
# TODO: Do we still need to do this?
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_id")] <-
  "geneID"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_name")] <-
  "geneName"
names(gtf@elementMetadata@listData)[which(names(gtf@elementMetadata@listData) == "gene_type")] <-
  "geneBiotype"
gtf@elementMetadata$id <- gtf@elementMetadata$geneID

genes <- gtf[which(gtf$type == "gene")]
gtfdf <- gtf %>% as.data.frame()
g2s <- as.data.frame(gtf@elementMetadata@listData) %>% 
  dplyr::select(geneID, geneName) %>% 
  distinct()
tx2gene <- gtfdf %>%
  mutate(TXNAME = transcript_id, GENEID = geneID) %>%
  dplyr::select(TXNAME, GENEID)
```



```{r}
# Read in sample annotations
samples <- read.csv("diff_samplesheet.csv")

samples_d2 <- samples %>% 
  filter(grepl("d2", Timepoint))
samples_d4 <- samples %>%
  filter(grepl("d4", Timepoint))
samples_d6 <- samples %>% 
  filter(grepl("d6", Timepoint))
samples_d10 <- samples %>% 
  filter(grepl("d10", Timepoint))
samples_d12 <- samples %>% 
  filter(grepl("d12", Timepoint))
rownames(samples_d2) <- samples_d2$SampleID
rownames(samples_d4) <- samples_d4$SampleID
rownames(samples_d6) <- samples_d6$SampleID
rownames(samples_d10) <- samples_d10$SampleID
rownames(samples_d12) <- samples_d12$SampleID

# Factorize condition variables
samples_d2$Timepoint <- factor(samples_d2$Condition,
                                 levels = c("CTL", "myh7b_KD"))
samples_d4$Timepoint <- factor(samples_d4$Condition,
                                levels = c("CTL", "myh7b_KD"))
samples_d6$Timepoint <- factor(samples_d6$Condition,
                                levels = c("CTL", "myh7b_KD"))
samples_d10$Timepoint <- factor(samples_d10$Condition,
                                levels = c("CTL", "myh7b_KD"))
samples_d12$Timepoint <- factor(samples_d12$Condition,
                                levels = c("CTL", "myh7b_KD"))


# Read in the salmon counts data.
files <- file.path("../results/salmon", samples_d2$SampleID, "quant.sf")
names(files) <- samples_d2$SampleID
txisalmon_d2 <- tximport(files, type = "salmon", tx2gene = tx2gene)

files <- file.path("../results/salmon", samples_d4$SampleID, "quant.sf")
names(files) <- samples_d4$SampleID
txisalmon_d4 <- tximport(files, type = "salmon", tx2gene = tx2gene)

files <- file.path("../results/salmon", samples_d6$SampleID, "quant.sf")
names(files) <- samples_d6$SampleID
txisalmon_d6 <- tximport(files, type = "salmon", tx2gene = tx2gene)

files <- file.path("../results/salmon", samples_d10$SampleID, "quant.sf")
names(files) <- samples_d10$SampleID
txisalmon_d10 <- tximport(files, type = "salmon", tx2gene = tx2gene)

files <- file.path("../results/salmon", samples_d12$SampleID, "quant.sf")
names(files) <- samples_d12$SampleID
txisalmon_d12 <- tximport(files, type = "salmon", tx2gene = tx2gene)
```

```{r retreive_tpm}
# TPM
tpm_d2 <- txisalmon_d2$abundance %>%
  as.data.frame() %>% 
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())
tpm_d4 <- txisalmon_d4$abundance %>%
  as.data.frame() %>% 
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())
tpm_d6 <- txisalmon_d6$abundance %>%
  as.data.frame() %>% 
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())
tpm_d10 <- txisalmon_d10$abundance %>%
  as.data.frame() %>% 
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())
tpm_d12 <- txisalmon_d12$abundance %>%
  as.data.frame() %>% 
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())

# Export count tables
write_csv(tpm_d2, "results/tpm_d2.csv")
write_csv(tpm_d4, "results/tpm_d4.csv")
write_csv(tpm_d6, "results/tpm_d6.csv")
write_csv(tpm_d10, "results/tpm_d10.csv")
write_csv(tpm_d12, "results/tpm_d12.csv")
```


```{r create_DESeqDataSet}
# Create DESeq2 experiment objects.
samples_d2 <- samples_d2[colnames(txisalmon_d2$counts),]
stopifnot(all(rownames(samples_d2) == colnames(txisalmon_d2$counts)))

samples_d4 <- samples_d4[colnames(txisalmon_d4$counts),]
stopifnot(all(rownames(samples_d4) == colnames(txisalmon_d4$counts)))

samples_d6 <- samples_d6[colnames(txisalmon_d6$counts),]
stopifnot(all(rownames(samples_d6) == colnames(txisalmon_d6$counts)))

samples_d10 <- samples_d10[colnames(txisalmon_d10$counts),]
stopifnot(all(rownames(samples_d10) == colnames(txisalmon_d10$counts)))

samples_d12 <- samples_d12[colnames(txisalmon_d12$counts),]
stopifnot(all(rownames(samples_d12) == colnames(txisalmon_d12$counts)))
# Let's also make sure the row ranges are in the same order
names(gtf) <- gtf$geneID

gtf <- gtf[rownames(txisalmon_d2$counts)]
dds_d2 <- DESeqDataSetFromTximport(txisalmon_d2,
                                        colData   = samples_d2,
                                        design    = ~ Condition,
                                        rowRanges = gtf)
gtf <- gtf[rownames(txisalmon_d4$counts)]
dds_d4 <- DESeqDataSetFromTximport(txisalmon_d4,
                                        colData   = samples_d4,
                                        design    = ~ Condition,
                                        rowRanges = gtf)
gtf <- gtf[rownames(txisalmon_d6$counts)]
dds_d6 <- DESeqDataSetFromTximport(txisalmon_d6,
                                        colData   = samples_d6,
                                        design    = ~ Condition,
                                        rowRanges = gtf)
gtf <- gtf[rownames(txisalmon_d10$counts)]
dds_d10 <- DESeqDataSetFromTximport(txisalmon_d10,
                                        colData   = samples_d10,
                                        design    = ~ Condition,
                                        rowRanges = gtf)
gtf <- gtf[rownames(txisalmon_d12$counts)]
dds_d12 <- DESeqDataSetFromTximport(txisalmon_d12,
                                        colData   = samples_d12,
                                        design    = ~ Condition,
                                        rowRanges = gtf)
```


```{r prefilter}
keep <- rowSums(counts(dds_d2)) >= 10
dds_d2 <- dds_d2[keep, ]
keep <- rowSums(counts(dds_d4)) >= 10
dds_d4 <- dds_d4[keep, ]
keep <- rowSums(counts(dds_d6)) >= 10
dds_d6 <- dds_d6[keep, ]
keep <- rowSums(counts(dds_d10)) >= 10
dds_d10 <- dds_d10[keep, ]
keep <- rowSums(counts(dds_d12)) >= 10
dds_d12 <- dds_d12[keep, ]
```


```{r deseq}
dds_d2 <- DESeq(dds_d2)
dds_d4 <- DESeq(dds_d4)
dds_d6 <- DESeq(dds_d6)
dds_d10 <- DESeq(dds_d10)
dds_d12 <- DESeq(dds_d12)

saveRDS(dds_d2, file = "dds_d2.rds")
saveRDS(dds_d4, file = "dds_d4.rds")
saveRDS(dds_d6, file = "dds_d6.rds")
saveRDS(dds_d10, file = "dds_d10.rds")
saveRDS(dds_d12, file = "dds_d12.rds")
```

```{r res_unshrunken}
res_d2 <- results(dds_d2) 
res_d4 <- results(dds_d4)
res_d6 <- results(dds_d6)
res_d10 <- results(dds_d10)
res_d12 <- results(dds_d12)

# Export
resdf_d2 <- res_d2 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())

resdf_d4 <- res_d4 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())

resdf_d6 <- res_d6 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())

resdf_d10 <- res_d10 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())

resdf_d12 <- res_d12 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s) %>%
  dplyr::select(geneID, geneName, everything())


# Just note that when the rmarkdown is knitted, it will consider
# The directory that it is saved in to be the working directory.
# For that reason, I usually set the working directory to be the 
# directory of the script that I'm currently working on.
# That way the knitting just works.
write_csv(resdf_d2, "results/res_unshrunken_d2.csv")
write_csv(resdf_d4, "results/res_unshrunken_d4.csv")
write_csv(resdf_d6, "results/res_unshrunken_d6.csv")
write_csv(resdf_d10, "results/res_unshrunken_d10.csv")
write_csv(resdf_d12, "results/res_unshrunken_d12.csv")
```

```{r res_shrunken}
# Use this to retrieve the coefficient.
#resultsNames(dds_d2)
res_shrunken_d2 <- lfcShrink(dds_d2, coef = "Condition_myh7b_KD_vs_CTL")
res_shrunken_d4 <- lfcShrink(dds_d4, coef = "Condition_myh7b_KD_vs_CTL")
res_shrunken_d6 <- lfcShrink(dds_d6, coef = "Condition_myh7b_KD_vs_CTL")
res_shrunken_d10 <- lfcShrink(dds_d10, coef = "Condition_myh7b_KD_vs_CTL")
res_shrunken_d12 <- lfcShrink(dds_d12, coef = "Condition_myh7b_KD_vs_CTL")

# Export
resdf_d2 <- res_shrunken_d2 %>% 
  as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s)

resdf_d4 <- res_shrunken_d4 %>% as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s)

resdf_d6 <- res_shrunken_d6 %>% as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s)

resdf_d10 <- res_shrunken_d10 %>% as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s)

resdf_d12 <- res_shrunken_d12 %>% as.data.frame() %>%
  rownames_to_column("geneID") %>%
  merge(g2s)

write_csv(resdf_d2, "results/res_shrunken_d2.csv")
write_csv(resdf_d4, "results/res_shrunken_d4.csv")
write_csv(resdf_d6, "results/res_shrunken_d6.csv")
write_csv(resdf_d10, "results/res_shrunken_d10.csv")
write_csv(resdf_d12, "results/res_shrunken_d12.csv")
```


# Plots

## PCA

Principal Component Analysis (PCA) is a statistical technique used to identify global patterns in high-dimensional datasets. It is commonly used to explore the similarity of biological samples in RNA-seq datasets. To achieve this, gene expression values are transformed into Principal Components (PCs), a set of linearly uncorrelated features which represent the most relevant sources of variance in the data, and subsequently visualized using a scatter plot.

Each point represents an RNA-seq sample. Samples with similar gene expression profiles are closer in the three-dimensional space. If provided, sample groups are indicated using different colors, allowing for easier interpretation of the results.


```{r variance_stabilization}
rld_d2 <- varianceStabilizingTransformation(dds_d2)
rld_d4 <- varianceStabilizingTransformation(dds_d4)
rld_d6 <- varianceStabilizingTransformation(dds_d6)
rld_d10 <- varianceStabilizingTransformation(dds_d10)
rld_d12 <- varianceStabilizingTransformation(dds_d12)
```


```{r plot_pca, message=FALSE}
DESeq2::plotPCA(rld_d2, intgroup = "Condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("d2 Differentiation PCA")
ggsave("figures/d2_pca.pdf")
ggsave("figures/d2_pca.png")

DESeq2::plotPCA(rld_d4, intgroup = "Condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("d4 Differentiation PCA")
ggsave("figures/d4_pca.pdf")
ggsave("figures/d4_pca.png")

DESeq2::plotPCA(rld_d6, intgroup = "Condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("d6 Differentiation PCA")
ggsave("figures/d6_pca.pdf")
ggsave("figures/d6_pca.png")

DESeq2::plotPCA(rld_d10, intgroup = "Condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("d10 Differentiation PCA")
ggsave("figures/d10_pca.pdf")
ggsave("figures/d10_pca.png")

DESeq2::plotPCA(rld_d12, intgroup = "Condition")  +
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle("d12 Differentiation PCA")
ggsave("figures/d12_pca.pdf")
ggsave("figures/d12_pca.png")
```

## Heatmap

This plot shows only differentially expressed genes on a per-sample basis. We have scaled the data by row and used the `ward.D2` method for clustering [@Ward1963-xf].

```{r plot_deg_heatmap}
plot_deg_heatmap <- function(res, rld, samp_df, interesting_groups = "Condition", 
                             padj_thresh = 0.01, lfc_thresh = 0, title,
                             ann_colors = list(condition = c(CTL  = "#424242", myh7b_KD = "#a8404c"))) 
  { deg <- res %>%
    as.data.frame() %>%
    rownames_to_column("geneID") %>%
    filter(padj < padj_thresh, abs(log2FoldChange) > lfc_thresh)
  
  rld_counts <- rld@assays@data[[1]]
  deg_rld_counts <- rld_counts[which(rownames(rld_counts) %in% deg$geneID),]
  rld_scaled <- t(scale(t(deg_rld_counts)))
  
  anndf <- samp_df %>% dplyr::select(interesting_groups)
  
  
  col_pal <- c(colorRampPalette(colors = c("#424242", "#ffffff"))(49),
               "#ffffff","#ffffff",
               colorRampPalette(colors = c("#ffffff", "#a8404c"))(49))
  
  levels(anndf$condition)
  htmap <- pheatmap(rld_scaled, color = col_pal,
                 show_rownames = F,
                 annotation_col = anndf,
                 annotation_colors = ann_colors,
                 main = title)
  return(htmap)
}

d2_heatmap <- plot_deg_heatmap(res_d2, rld_d2, samp_df = samples_d2, interesting_groups = "Condition",
                 padj_thresh = 0.01, lfc_thresh = 0, title = "d2 DEG")
pdf("figures/d2_deg_heatmap.pdf"); show(d2_heatmap); dev.off()
png("figures/d2_deg_heatmap.png"); show(d2_heatmap); dev.off()

d4_heatmap <- plot_deg_heatmap(res_d4, rld_d4, samp_df = samples_d4, interesting_groups = "Condition",
                 padj_thresh = 0.01, lfc_thresh = 0, title = "d4 DEG")
pdf("figures/d4_deg_heatmap.pdf"); show(d4_heatmap); dev.off()
png("figures/d4_deg_heatmap.png"); show(d4_heatmap); dev.off()

d6_heatmap <- plot_deg_heatmap(res_d6, rld_d6, samp_df = samples_d6, interesting_groups = "Condition",
                 padj_thresh = 0.01, lfc_thresh = 0, title = "d6 DEG")
pdf("figures/d6_deg_heatmap.pdf"); show(d6_heatmap); dev.off()
png("figures/d6_deg_heatmap.png"); show(d6_heatmap); dev.off()

d10_heatmap <- plot_deg_heatmap(res_d10, rld_d10, samp_df = samples_d10, interesting_groups = "Condition",
                 padj_thresh = 0.01, lfc_thresh = 0, title = "d10 DEG")
pdf("figures/d10_deg_heatmap.pdf"); show(d10_heatmap); dev.off()
png("figures/d10_deg_heatmap.png"); show(d10_heatmap); dev.off()

d12_heatmap <- plot_deg_heatmap(res_d12, rld_d12, samp_df = samples_d12, interesting_groups = "Condition",
                 padj_thresh = 0.01, lfc_thresh = 0, title = "d12 DEG")
pdf("figures/d12_deg_heatmap.pdf"); show(d12_heatmap); dev.off()
png("figures/d12_deg_heatmap.png"); show(d12_heatmap); dev.off()

```


```{r}
# MA plots
g <- ggplot(resdf_d2, aes(x = log10(baseMean), y = log2FoldChange, label = geneName))
g + geom_hline(yintercept = 0) +
  geom_point(data = resdf_d2 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d2 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.8) + 
  geom_text_repel(data = resdf_d2 %>% filter(padj < 0.01)) +
  ggtitle("d2")
ggsave("figures/d2_maplot.pdf")
ggsave("figures/d2_maplot.png")

g <- ggplot(resdf_d4, aes(x = log10(baseMean), y = log2FoldChange, label = geneName))
g + geom_hline(yintercept = 0) +
  geom_point(data = resdf_d4 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d4 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.8) + 
  geom_text_repel(data = resdf_d4 %>% filter(padj < 0.01)) +
  ggtitle("d4")
ggsave("figures/d4_maplot.pdf")
ggsave("figures/d4_maplot.png")

g <- ggplot(resdf_d6, aes(x = log10(baseMean), y = log2FoldChange, label = geneName))
g + geom_hline(yintercept = 0) +
  geom_point(data = resdf_d6 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d6 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.8) + 
  geom_text_repel(data = resdf_d6 %>% filter(abs(log2FoldChange) > 1, padj < 0.0001)) +
  ggtitle("d6")
ggsave("figures/d6_maplot.pdf")
ggsave("figures/d6_maplot.png")

g <- ggplot(resdf_d10, aes(x = log10(baseMean), y = log2FoldChange, label = geneName))
g + geom_hline(yintercept = 0) +
  geom_point(data = resdf_d10 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d10 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.8) + 
  geom_text_repel(data = resdf_d10 %>% filter(abs(log2FoldChange) > 3.5, padj < 0.000000001)) +
  ggtitle("d10")
ggsave("figures/d10_maplot.pdf")
ggsave("figures/d10_maplot.png")

g <- ggplot(resdf_d12, aes(x = log10(baseMean), y = log2FoldChange, label = geneName))
g + geom_hline(yintercept = 0) +
  geom_point(data = resdf_d12 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d12 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.8) + 
  geom_text_repel(data = resdf_d12 %>% filter(abs(log2FoldChange) > 4, padj < 0.000000001)) +
  ggtitle("d12")
ggsave("figures/d12_maplot.pdf")
ggsave("figures/d12_maplot.png")
```


```{r}
# Volcano plots 
g <- ggplot(resdf_d2, aes(x = log2FoldChange, y = -log10(padj), label = geneName))
g + geom_vline(xintercept = 0, lty = 2) +
  geom_point(data = resdf_d2 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d2 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.6) + 
  geom_text_repel(data = resdf_d2 %>% filter(padj < 0.01)) +
  ggtitle("d2")
ggsave("figures/d2_volcano.pdf")
ggsave("figures/d2_volcano.png")

g <- ggplot(resdf_d4, aes(x = log2FoldChange, y = -log10(padj), label = geneName))
g + geom_vline(xintercept = 0, lty = 2) +
  geom_point(data = resdf_d4 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d4 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.6) + 
  geom_text_repel(data = resdf_d4 %>% filter(padj < 0.01)) +
  ggtitle("d4")
ggsave("figures/d4_volcano.pdf")
ggsave("figures/d4_volcano.png")

g <- ggplot(resdf_d6, aes(x = log2FoldChange, y = -log10(padj), label = geneName))
g + geom_vline(xintercept = 0, lty = 2) +
  geom_point(data = resdf_d6 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d6 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.6) + 
  geom_text_repel(data = resdf_d6 %>% filter(abs(log2FoldChange) > 1, padj < 0.0001)) +
  ggtitle("d6")
ggsave("figures/d6_volcano.pdf")
ggsave("figures/d6_volcano.png")

g <- ggplot(resdf_d10, aes(x = log2FoldChange, y = -log10(padj), label = geneName))
g + geom_vline(xintercept = 0, lty = 2) +
  geom_point(data = resdf_d10 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d10 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.6) + 
  geom_text_repel(data = resdf_d10 %>% filter(abs(log2FoldChange) > 3.5, padj < 0.000000001)) +
  ggtitle("d10")
ggsave("figures/d10_volcano.pdf")
ggsave("figures/d10_volcano.png")

g <- ggplot(resdf_d12, aes(x = log2FoldChange, y = -log10(padj), label = geneName))
g + geom_vline(xintercept = 0, lty = 2) +
  geom_point(data = resdf_d12 %>% filter(padj >= 0.01), color = "#424242", alpha = 0.4) + 
  geom_point(data = resdf_d12 %>% filter(padj < 0.01), color = "#a8404c", alpha = 0.6) + 
  geom_text_repel(data = resdf_d12 %>% filter(abs(log2FoldChange) > 4, padj < 0.000000001)) +
  ggtitle("d12")
ggsave("figures/d12_volcano.pdf")
ggsave("figures/d12_volcano.png")
```


## MYH7b plot

This is the gene we intended to knockdown, so let's just see what this gene looks like.

```{r MYH7B}
gene_name <- "MYH7B"
gene_id <- g2s[which(g2s$geneName == gene_name),"geneID"] 
gene_tpm_d2 <- tpm_d2[which(tpm_d2$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d2)
  
g <- ggplot(gene_tpm_d2, aes(x = 
                               Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d2 -- ", gene_name, " expression"))
ggsave(paste0("figures/d2_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d2_", gene_name, "_tpm.png"))

gene_tpm_d4 <- tpm_d4[which(tpm_d4$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d4)

g <- ggplot(gene_tpm_d4, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d4 -- ", gene_name, " expression"))
ggsave(paste0("figures/d4_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d4_", gene_name, "_tpm.png"))

gene_tpm_d6 <- tpm_d6[which(tpm_d6$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d6)

g <- ggplot(gene_tpm_d6, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d6 -- ", gene_name, " expression"))
ggsave(paste0("figures/d6_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d6_", gene_name, "_tpm.png"))

gene_tpm_d10 <- tpm_d10[which(tpm_d10$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d10)

g <- ggplot(gene_tpm_d10, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d10 -- ", gene_name, " expression"))
ggsave(paste0("figures/d10_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d10_", gene_name, "_tpm.png"))

gene_tpm_d12 <- tpm_d12[which(tpm_d12$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d12)

g <- ggplot(gene_tpm_d12, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d12 -- ", gene_name, " expression"))
ggsave(paste0("figures/d12_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d12_", gene_name, "_tpm.png"))
```

```{r MYH7}
gene_name <- "MYH7"
gene_id <- g2s[which(g2s$geneName == gene_name),"geneID"] 
gene_tpm_d2 <- tpm_d2[which(tpm_d2$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d2)
  
g <- ggplot(gene_tpm_d2, aes(x = 
                               Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d2 -- ", gene_name, " expression"))
ggsave(paste0("figures/d2_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d2_", gene_name, "_tpm.png"))

gene_tpm_d4 <- tpm_d4[which(tpm_d4$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d4)

g <- ggplot(gene_tpm_d4, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d4 -- ", gene_name, " expression"))
ggsave(paste0("figures/d4_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d4_", gene_name, "_tpm.png"))

gene_tpm_d6 <- tpm_d6[which(tpm_d6$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d6)

g <- ggplot(gene_tpm_d6, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d6 -- ", gene_name, " expression"))
ggsave(paste0("figures/d6_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d6_", gene_name, "_tpm.png"))

gene_tpm_d10 <- tpm_d10[which(tpm_d10$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d10)

g <- ggplot(gene_tpm_d10, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d10 -- ", gene_name, " expression"))
ggsave(paste0("figures/d10_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d10_", gene_name, "_tpm.png"))

gene_tpm_d12 <- tpm_d12[which(tpm_d12$geneID == gene_id), ] %>%
  pivot_longer(3:ncol(.), names_to = "sample_id", values_to = "tpm") %>%
  merge(samples_d12)

g <- ggplot(gene_tpm_d12, aes(x = Condition, y = tpm, color = Condition))
g + geom_point() +
  stat_summary(fun = mean, fun.min = mean, fun.max = mean,
               geom = "crossbar", width = 0.5) + 
  scale_color_manual(values = c("#424242","#a8404c")) + 
  ggtitle(paste0("d12 -- ", gene_name, " expression"))
ggsave(paste0("figures/d12_", gene_name, "_tpm.pdf"))
ggsave(paste0("figures/d12_", gene_name, "_tpm.png"))
```



#```{r}
# Fold change correlation between experiments
res_combined_wide <- res_combined %>%
  dplyr::select(geneID, geneName, log2FoldChange, padj, experiment) %>%
  pivot_wider(id_cols = c("geneID", "geneName"), names_from = experiment,
              values_from = c("log2FoldChange", "padj")) %>%
  mutate(sig_d4 = padj_d4 < 0.01,
         sig_d2 = padj_d2 < 0.01,
         sig_both = sig_d4 & sig_d2)
# Label points
res_combined_wide$sig <- "NS"
res_combined_wide[which(res_combined_wide$sig_d2), "sig"] <- "DE in d2"
res_combined_wide[which(res_combined_wide$sig_d4), "sig"] <- "DE in d4"
res_combined_wide[which(res_combined_wide$sig_d6), "sig"] <- "DE in d6"
res_combined_wide[which(res_combined_wide$sig_d10), "sig"] <- "DE in d10"
res_combined_wide[which(res_combined_wide$sig_d12), "sig"] <- "DE in d12"
res_combined_wide[which(res_combined_wide$sig_both), "sig"] <- "DE in all"
res_combined_wide$sig <- factor(res_combined_wide$sig, levels = c("NS", "DE in d2", "DE in d4", "DE in d6", "DE in d10", "DE in d12", "DE in all"))


# NOTE: This is using shrunken log2FoldChanges from DESeq2.
g <- ggplot(res_combined_wide, aes(x = log2FoldChange_d4, y = log2FoldChange_d2, color = sig))
g + geom_hline(color = "#212121", yintercept = 0, lty = 2) + 
  geom_vline(color = "#212121", xintercept = 0, lty = 2) +
  scale_color_manual(values = c("#424242","#71969F","#8B1D3B", "#DD1822"), name = "padj < 0.01") +
  geom_point(alpha = 0.7) + 
  theme_paperwhite() + 
  stat_cor() + 
  ggtitle("IPSCM: ASO KD vs SP OE")
ggsave("figures/ipscm_scatter.png")
ggsave("figures/ipscm_scatter.pdf")

# Let's export this handy table
write_csv(res_combined_wide, "d2_d4_shrunken_lfc_compared.csv")
```

