---
---
title: "Functional Analysis: lncMYH7b Knockdown RNA-seq in iPSCMs, Novogene"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
bibliography: bibliography.bib
params:
  # DESeqDataSet object
  # DESeqResults object
  dds_file: !r file.path("../data", Sys.Date(), "dds.rda")
  res_file: !r file.path("../data", Sys.Date(), "res_list_shrunken.rda")
  organism: "Homo sapiens"
  go_class: "BP"
  lfc_threshold: 0.5
  data_dir: !r file.path("../data", Sys.Date())
  results_dir: !r file.path("../results", Sys.Date(), "functional_analysis")
title: "Functional Analysis: lncMYH7b Knockdown RNA-seq in iPSCMs, Novogene"  
editor_options: 
  chunk_output_type: console
---

```{r setup, message=FALSE}
options(stringsAsFactors = TRUE)

# Additional required packages =================================================
# KEGG supported organisms
# http://www.genome.jp/kegg/catalog/org_list.html

# OrgDb supported organisms
# http://bioconductor.org/packages/release/BiocViews.html#___OrgDb

library(clusterProfiler)
library(DOSE)
library(pathview)
library(org.Hs.eg.db)
library(tidyverse)
org_db <- "org.Hs.eg.db"

```


```{r}
dds_d2 <- readRDS("results/dds_d2.rds")
dds_d4 <- readRDS("results/dds_d4.rds")
dds_d6 <- readRDS("results/dds_d6.rds")
dds_d10 <- readRDS("results/dds_d10.rds")
dds_d12 <- readRDS("results/dds_d12.rds")


# Let's add in the Enembl gene IDs.
ens2eg <- as.list(org.Hs.egENSEMBL2EG)

ensemblids<- unlist(lapply(as.character(rownames(rowData(dds_d2))), 
                           function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]] }))
rowData(dds_d2)$entrezID <- ens2eg[match(ensemblids, names(ens2eg))]

ensemblids<- unlist(lapply(as.character(rownames(rowData(dds_d4))), 
                           function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]] }))
rowData(dds_d4)$entrezID <- ens2eg[match(ensemblids, names(ens2eg))]

ensemblids<- unlist(lapply(as.character(rownames(rowData(dds_d6))), 
                           function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]] }))
rowData(dds_d6)$entrezID <- ens2eg[match(ensemblids, names(ens2eg))]

ensemblids<- unlist(lapply(as.character(rownames(rowData(dds_d10))), 
                           function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]] }))
rowData(dds_d10)$entrezID <- ens2eg[match(ensemblids, names(ens2eg))]

ensemblids<- unlist(lapply(as.character(rownames(rowData(dds_d12))), 
                           function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]] }))
rowData(dds_d12)$entrezID <- ens2eg[match(ensemblids, names(ens2eg))]



kegg_code <- "hsa"

# Check for valid GO class
# BP: Biological Process
# CC: Cellular Component
# MF: Molecular Function
go_classes <- c("BP", "CC", "MF")

```


```{r results}
padj_thresh <- 0.01
l2fc_thresh <- 0

res_d2 <- DESeq2::results(dds_d2) 
res_d2_df <- res_d2 %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(padj))
write.csv(res_d2_df, file = "res_d2_df.csv")
res_d4 <- DESeq2::results(dds_d4) 
res_d4_df <- res_d4 %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(padj))
write.csv(res_d4_df, file = "res_d4_df.csv")
res_d6 <- DESeq2::results(dds_d6) 
res_d6_df <- res_d6 %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(padj))
write.csv(res_d6_df, file = "res_d6_df.csv")
res_d10 <- DESeq2::results(dds_d10) 
res_d10_df <- res_d10 %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(padj))
write.csv(res_d10_df, file = "res_d10_df.csv")
res_d12 <- DESeq2::results(dds_d12) 
res_d12_df <- res_d12 %>%
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(!is.na(padj))
write.csv(res_d12_df, file = "res_d12_df.csv")


# Use the all_genes vector for background (aka universe)
all_genes_d2 <- sapply(rownames(res_d2), function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(all_genes_d2) <- NULL

all_genes_d4 <- sapply(rownames(res_d4), function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(all_genes_d4) <- NULL

all_genes_d6 <- sapply(rownames(res_d6), function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(all_genes_d6) <- NULL

all_genes_d10 <- sapply(rownames(res_d10), function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(all_genes_d10) <- NULL

all_genes_d12 <- sapply(rownames(res_d12), function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(all_genes_d12) <- NULL

# help("significants", "DEGreport")
sig_d2_df <- res_d2_df %>% 
  filter(padj < padj_thresh, abs(log2FoldChange) > l2fc_thresh)
write.csv(sig_d2_df, "results/sig_d2.csv")
sig_d4_df <- res_d4_df %>%
  filter(padj < padj_thresh, abs(log2FoldChange) > l2fc_thresh)
write.csv(sig_d4_df, "results/sig_d4.csv")
sig_d6_df <- res_d6_df %>%
  filter(padj < padj_thresh, abs(log2FoldChange) > l2fc_thresh)
write.csv(sig_d6_df, "results/sig_d6.csv")
sig_d10_df <- res_d10_df %>%
  filter(padj < padj_thresh, abs(log2FoldChange) > l2fc_thresh)
write.csv(sig_d10_df, "results/sig_d10.csv")
sig_d12_df <- res_d12_df %>%
  filter(padj < padj_thresh, abs(log2FoldChange) > l2fc_thresh)
write.csv(sig_d12_df, "results/sig_d12.csv")

sig_genes_d2 <- sapply(sig_d2_df$gene_id, function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(sig_genes_d2) <- NULL
sig_genes_d4 <- sapply(sig_d4_df$gene_id, function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(sig_genes_d4) <- NULL
sig_genes_d6 <- sapply(sig_d6_df$gene_id, function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(sig_genes_d6) <- NULL
sig_genes_d10 <- sapply(sig_d10_df$gene_id, function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(sig_genes_d10) <- NULL
sig_genes_d12 <- sapply(sig_d12_df$gene_id, function(x) { unlist(base::strsplit(x, ".", fixed = T))[[1]]})
names(sig_genes_d12) <- NULL

lfc_vec_d2 <- sig_d2_df$log2FoldChange
lfc_vec_d4 <- sig_d4_df$log2FoldChange
lfc_vec_d6 <- sig_d6_df$log2FoldChange
lfc_vec_d10 <- sig_d10_df$log2FoldChange
lfc_vec_d12 <- sig_d12_df$log2FoldChange
names(lfc_vec_d2) <- sig_d2_df$gene_id
names(lfc_vec_d4) <- sig_d4_df$gene_id
names(lfc_vec_d6) <- sig_d6_df$gene_id
names(lfc_vec_d10) <- sig_d10_df$gene_id
names(lfc_vec_d12) <- sig_d12_df$gene_id
# Sort from upregulated to downregulated
lfc_vec_d2 <- sort(lfc_vec_d2, decreasing = TRUE)
lfc_vec_d4 <- sort(lfc_vec_d4, decreasing = TRUE)
lfc_vec_d6 <- sort(lfc_vec_d6, decreasing = TRUE)
lfc_vec_d10 <- sort(lfc_vec_d10, decreasing = TRUE)
lfc_vec_d12 <- sort(lfc_vec_d12, decreasing = TRUE)
```

## GO enrichment analysis

```{r enrich_go}
# This will be a lot of duplicated code, so let's do this in a loop
experiments <- c("d2", "d4", "d6")
sig_genes <- list("d2" = sig_genes_d2, "d4" = sig_genes_d4, "d6" = sig_genes_d6)
all_genes <- list("d2" = all_genes_d2, "d4" = all_genes_d4, "d6" = all_genes_d6)

for(i in 1:length(experiments)) {
  for(j in 1:length(go_classes)) {
    
    enrich_go <- enrichGO(gene = sig_genes[[experiments[i]]], 
                          OrgDb = org_db,
                          keyType = "ENSEMBL",
                          ont = go_classes[j],
                          universe = all_genes[[experiments[i]]],
                          qvalueCutoff = 1,
                          readable = TRUE,
                          minGSSize = 4)
    enrich_go_tbl <- enrich_go@result %>%
      as_tibble()
    
    write_csv(enrich_go_tbl, 
              path = file.path("results",
                               paste0(paste(experiments[i], "go", go_classes[j], 
                                            "clusterprofiler", "padj", padj_thresh,
                                            "lfc", l2fc_thresh, sep = "_"), ".csv")))
    enrich_go_tbl
    
    dp <- dotplot(enrich_go, showCategory = 25) +
      ggtitle(paste0(go_classes[j], ": ", experiments[i]))
    show(dp)
    ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_dotplot.png"))
    ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_dotplot.pdf"))
    
    if(length(which(enrich_go@result$qvalue < 0.05)) > 0) {
      em <- emapplot(enrich_go, showCategory = 25) +
        ggtitle(paste0(go_classes[1], ": ", experiments[i]))
      show(em)
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_emapplot.png"))
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_emapplot.pdf"))
    }
    
    if(length(which(enrich_go@result$qvalue < 0.05)) > 0) {
      cn <- cnetplot(enrich_go, foldChange = lfc_vec_d2) + 
        ggtitle(paste0(go_classes[1], ": ", experiments[i]))
      show(cn)
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_cnetplot.png"))
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_cnetplot.pdf"))
    }
  }
}
```



```{r enrich_go}
# This will be a lot of duplicated code, so let's do this in a loop
experiments <- c("d10", "d12")
sig_genes <- list("d10" = sig_genes_d10, "d12" = sig_genes_d12)
all_genes <- list("d10" = all_genes_d10, "d12" = all_genes_d12)

for(i in 1:length(experiments)) {
  for(j in 1:length(go_classes)) {
    
    enrich_go <- enrichGO(gene = sig_genes[[experiments[i]]], 
                          OrgDb = org_db,
                          keyType = "ENSEMBL",
                          ont = go_classes[j],
                          universe = all_genes[[experiments[i]]],
                          qvalueCutoff = 1,
                          readable = TRUE,
                          minGSSize = 4)
    enrich_go_tbl <- enrich_go@result %>%
      as_tibble()
    
    write_csv(enrich_go_tbl, 
              path = file.path("results",
                               paste0(paste(experiments[i], "go", go_classes[j], 
                                            "clusterprofiler", "padj", padj_thresh,
                                            "lfc", l2fc_thresh, sep = "_"), ".csv")))
    enrich_go_tbl
    
    dp <- dotplot(enrich_go, showCategory = 25) +
      ggtitle(paste0(go_classes[j], ": ", experiments[i]))
    show(dp)
    ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_dotplot.png"))
    ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_dotplot.pdf"))
    
    if(length(which(enrich_go@result$qvalue < 0.05)) > 0) {
      em <- emapplot(enrich_go, showCategory = 25) +
        ggtitle(paste0(go_classes[1], ": ", experiments[i]))
      show(em)
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_emapplot.png"))
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_emapplot.pdf"))
    }
    
    if(length(which(enrich_go@result$qvalue < 0.05)) > 0) {
      cn <- cnetplot(enrich_go, foldChange = lfc_vec_d10) + 
        ggtitle(paste0(go_classes[1], ": ", experiments[i]))
      show(cn)
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_cnetplot.png"))
      ggsave(paste0("figures/", experiments[i],"_go", go_classes[j], "_cnetplot.pdf"))
    }
  }
}
```


## GO GSEA analysis

A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The enrichment analysis we demonstrated previously were based on these differentially expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. [Gene Set Enrichment Analysis (GSEA)][GSEA] directly addresses this limitation. All genes can be used in [GSEA][]; [GSEA][] aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.

```{r gsea_go}
# Prepare the gene list.
# Here we're subtracting the adjusted P value from 1.
res_d2_df <- read.csv("res_d2_df.csv")
res_d2_df$X = NULL
row.names(res_d2_df) <- res_d2_df$gene_id
gene_list_d2 <- res_d2_df$log2FoldChange
names(gene_list_d2) <- row.names(res_d2_df)
names(gene_list_d2) <- sapply(names(gene_list_d2), 
                           function(x) {unlist(base::strsplit(x, ".", fixed = T))[[1]]})
gene_list_d2 <- sort(gene_list_d2, decreasing = TRUE)
summary(gene_list_d2)


# Now run GSEA
#help("gseGO", "clusterProfiler")
# minGSSize = 100
gsea_go_d2 <- gseGO(
    geneList = gene_list_d2,
    ont = "BP",
    OrgDb = org_db,
    keyType = "ENSEMBL",
    pvalueCutoff = 1,
    minGSSize = 10
)
print(gsea_go_d2)
gsea_go_tbld2 <- gsea_go_d2 %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go_d2, gsea_go_tbld2)
write_csv(gsea_go_tbld2, "results/gsea_go_tbld2.csv")

res_d4_df <- read.csv("res_d4_df.csv")
res_d4_df$X = NULL
row.names(res_d4_df) <- res_d4_df$gene_id
gene_list_d4 <- res_d4_df$log2FoldChange
names(gene_list_d4) <- row.names(res_d4_df)
names(gene_list_d4) <- sapply(names(gene_list_d4), 
                           function(x) {unlist(base::strsplit(x, ".", fixed = T))[[1]]})
gene_list_d4 <- sort(gene_list_d4, decreasing = TRUE)
summary(gene_list_d4)


# Now run GSEA
#help("gseGO", "clusterProfiler")
# minGSSize = 100
gsea_go_d4 <- gseGO(
    geneList = gene_list_d4,
    ont = "BP",
    OrgDb = org_db,
    keyType = "ENSEMBL",
    pvalueCutoff = 1,
    minGSSize = 10
)
print(gsea_go_d4)
gsea_go_tbld4 <- gsea_go_d4 %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go_d4, gsea_go_tbld4)
write_csv(gsea_go_tbld4, "results/gsea_go_tbld4.csv")

res_d6_df <- read.csv("res_d6_df.csv")
res_d6_df$X = NULL
row.names(res_d6_df) <- res_d6_df$gene_id
gene_list_d6 <- res_d6_df$log2FoldChange
names(gene_list_d6) <- row.names(res_d6_df)
names(gene_list_d6) <- sapply(names(gene_list_d6), 
                           function(x) {unlist(base::strsplit(x, ".", fixed = T))[[1]]})
gene_list_d6 <- sort(gene_list_d6, decreasing = TRUE)
summary(gene_list_d6)


# Now run GSEA
#help("gseGO", "clusterProfiler")
# minGSSize = 100
gsea_go_d6 <- gseGO(
    geneList = gene_list_d6,
    ont = "BP",
    OrgDb = org_db,
    keyType = "ENSEMBL",
    pvalueCutoff = 1,
    minGSSize = 10
)
print(gsea_go_d6)
gsea_go_tbld6 <- gsea_go_d6 %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go_d6, gsea_go_tbld6)
write_csv(gsea_go_tbld6, "results/gsea_go_tbld6.csv")

res_d10_df <- read.csv("res_d10_df.csv")
res_d10_df$X = NULL
row.names(res_d10_df) <- res_d10_df$gene_id
gene_list_d10 <- res_d10_df$log2FoldChange
names(gene_list_d10) <- row.names(res_d10_df)
names(gene_list_d10) <- sapply(names(gene_list_d10), 
                           function(x) {unlist(base::strsplit(x, ".", fixed = T))[[1]]})
gene_list_d10 <- sort(gene_list_d10, decreasing = TRUE)
summary(gene_list_d10)


# Now run GSEA
#help("gseGO", "clusterProfiler")
# minGSSize = 100
gsea_go_d10 <- gseGO(
    geneList = gene_list_d10,
    ont = "BP",
    OrgDb = org_db,
    keyType = "ENSEMBL",
    pvalueCutoff = 1,
    minGSSize = 10
)
print(gsea_go_d10)
gsea_go_tbld10 <- gsea_go_d10 %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go_d10, gsea_go_tbld10)
write_csv(gsea_go_tbld10, "results/gsea_go_tbld10.csv")

res_d12_df <- read.csv("res_d12_df.csv")
res_d12_df$X = NULL
row.names(res_d12_df) <- res_d12_df$gene_id
gene_list_d12 <- res_d12_df$log2FoldChange
names(gene_list_d12) <- row.names(res_d12_df)
names(gene_list_d12) <- sapply(names(gene_list_d12), 
                           function(x) {unlist(base::strsplit(x, ".", fixed = T))[[1]]})
gene_list_d12 <- sort(gene_list_d12, decreasing = TRUE)
summary(gene_list_d12)


# Now run GSEA
#help("gseGO", "clusterProfiler")
# minGSSize = 100
gsea_go_d12 <- gseGO(
    geneList = gene_list_d12,
    ont = "BP",
    OrgDb = org_db,
    keyType = "ENSEMBL",
    pvalueCutoff = 1,
    minGSSize = 10
)
print(gsea_go_d12)
gsea_go_tbld12 <- gsea_go_d12 %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_go_d12, gsea_go_tbld12)
write_csv(gsea_go_tbld12, "results/gsea_go_tbld12.csv")

```

# KEGG enrichment analysis

## Map gene IDs to Entrez IDs

[Entrez][] IDs are required for [Kyoto Encyclopedia of Genes and Genomes (KEGG)][KEGG] analysis. Here we are defining 1:1 mappings of the [Ensembl][] gene IDs to [Entrez][] IDs. For genes that map to multiple [Entrez][] IDs, we are using the oldest [Entrez][] ID to define the 1:1 mapping.

```{r gene2entrez}
# Define the Ensembl gene to Entrez identifier mappings
gene2entrez <- rowData(dds_d2)$entrezID
stopifnot(is.list(gene2entrez))
stopifnot(!is.null(names(gene2entrez)))
# For genes that don't map 1:1 with Entrez, use the oldest Entrez ID.
# Also, ensure the IDs are coerced to character.
gene2entrez <- lapply(gene2entrez, function(x) {
    if (all(is.na(x))) {
        NULL
    } else {
        as.character(sort(x)[[1L]])
    }
})
# Drop genes that don't match to Entrez
gene2entrez <- Filter(Negate(is.null), gene2entrez)
# Ensembl gene IDs are names; Entrez gene ID are values
gene2entrez <- unlist(gene2entrez)
glimpse(gene2entrez)
saveData(gene2entrez)
```

Now let's get the background of unique Entrez IDs.

```{r all_entrez}
# Unique Entrez IDs to use for background
all_entrez <- gene2entrez %>%
    unique() %>%
    as.integer() %>%
    sort() %>%
    as.character()
glimpse(all_entrez)
```

Now we need to map the DESeqResults values to our Entrez IDs. Let's obtain a vector of significant genes that map to Entrez.

```{r sig_gene2entrez}
# Significant genes that map to Entrez ID (which can be duplicated)gene2entrez %>%

sig_d2_df <- read.csv("results/sig_d2.csv")
sig_d2_df$X = NULL
sig_gened2entrez <- intersect(names(gene2entrez), sig_d2_df$gene_id)
glimpse(sig_gened2entrez)
# any(duplicated(sig_gene2entrez))
```

Note that this vector can contain duplicate Entrez IDs, which we will resolve by significance (adjusted P value).

```{r entrez_res}
rownames(res_d2_df) <- sapply(rownames(res_d2_df), function(x) {
  unlist(strsplit(x, ".", fixed = T))[[1]]
})
stopifnot(all(names(sig_gene2entrez) %in% rownames(res_d2_df)))

# Filter to contain only significant genes that map to Entrez
sig_entrez_res_df <- res_d2_df %>%
    .[names(sig_gene2entrez), ] %>%
    rownames_to_column("geneID") %>%
    mutate(entrezID = sig_gene2entrez) %>%
    group_by(entrezID) %>%
    # Get the most significant result, per Entrez ID
    top_n(n = 1, wt = padj) %>%
    as.data.frame() %>%
    set_rownames(.$entrezID)
saveData(sig_entrez_res_df)
sig_entrez <- pull(sig_entrez_res_df, entrezID)

# Extract the fold changes
entrez_lfc_vec <- sig_entrez_res_df$log2FoldChange
names(entrez_lfc_vec) <- sig_entrez_res_df$entrezID
# Sort from upregulated to downregulated
entrez_lfc_vec <- sort(entrez_lfc_vec, decreasing = TRUE)
glimpse(entrez_lfc_vec)
```

```{r enrich_kegg}
kegg <- enrichKEGG(
    gene = as.character(sig_entrez),
    organism = kegg_code,
    keyType = "ncbi-geneid",
    universe = as.character(all_entrez),
    qvalueCutoff = 0.05
)
print(kegg)
kegg_tbl <- kegg %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(kegg, kegg_tbl, dir = params$data_dir)
write_csv(
    kegg_tbl,
    path = file.path(
        params$results_dir,
        paste0(
            paste(
                "kegg",
                "clusterprofiler",
                "padj",
                alpha,
                "lfc",
                params$lfc_threshold,
                sep = "_"
            ),
            ".csv.gz"
        )
    )
)
kegg_tbl
```

## KEGG GSEA analysis

[GSEA][] analysis is performed with the [clusterProfiler][] tool using KEGG gene sets and using the log2 fold changes as input. By using the log2 fold changes as the input, we are identifying pathways with genes that exhibit coordinated fold changes that are larger than might be expected by chance. The significant pathways can be visualized using the log2 fold changes with the Pathview tool.

Gene set enrichment analysis tools use ranked lists of genes (here ranked by log2FC) without using a threshold. This allows the tools to use more information to identify enriched biological processes. The [introduction to gene set enrichment analysis](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896) goes into more detail about some of the advantages of this approach. By using the log2 fold changes as the input, we are identifying pathways with genes that exhibit coordinated fold changes that are larger than might be expected by chance. The significant pathways can be visualized using the log2 fold changes with the [pathview][] tool.

The significantly dysregulated pathways (q-value (FDR) < 0.05) are displayed below in the pathway images, which show the degree of dysregulation of the genes (the minus direction (green) is down-regulated, while the positive direction (red) is up-regulated).

When performing [GSEA][] analysis it may be useful to adjust the minGSSize and/or maxGSSize parameter based on the pathways you would like to search for significance. If you are interested in smaller pathways, such as phototransduction, which has a gene set size of 24 genes, then you would want to adjust the minGSSize to less than 24. If you are only interested in larger pathways, then you would want to adjust the GSSize to a larger number. The fewer pathways tested, the less we need to correct for multiple test correction, so by adjusting the minGSSize and maxGSSize parameters you can test fewer pathways and limit testing to the pathways of interest.

```{r kegg_gsea}
gsea_kegg <- gseKEGG(
    geneList = entrez_lfc_vec,
    organism = kegg_code,
    keyType = "ncbi-geneid"
)
print(gsea_kegg)
gsea_kegg_tbl <- gsea_kegg %>%
    slot("result") %>%
    as_tibble() %>%
    camel()
saveData(gsea_kegg, dir = params$data_dir)
write_csv(
    gsea_kegg_tbl,
    path = file.path(
        params$results_dir, 
        paste0(
            paste(
                "gsea",
                "kegg",
                "clusterprofiler",
                "padj",
                alpha,
                "lfc",
                params$lfc_threshold,
                sep = "_"
            ),
            ".csv.gz"
        )
    )
)
gsea_kegg_tbl
```

```{r kegg_plots, message=FALSE, results="asis"}
# help("pathview", "pathview")
# 
# There is currently no way to set the output path of the pathview PNG files, so
# we're changing the working directory. Generally this is not recommended!
#
# Also, We're using `tryCatch()` here to return to the user any pathways that
# didn't output graphics correctly.
pathways <- gsea_kegg_tbl$id
if (length(pathways)) {
    # dplyr must be unloaded for pathview to work
    suppressWarnings(detach("package:dplyr", unload = TRUE, force = TRUE))
    wd <- getwd()
    setwd(kegg_plots_dir)
    invisible(lapply(pathways, function(pathway) {
        # `gene.data` vector should be numeric with Entrez IDs as names
        tryCatch(
            pathview(
                gene.data = entrez_lfc_vec,
                pathway.id = pathway,
                species = kegg_code, 
                limit = list(gene = 2, cpd = 1)
            ),
            error = function(e) {
                # Return a warning instead of an error
                warning(paste(pathway, "failed to plot"), call. = FALSE)
            }
        )
    }))
    setwd(wd)
    figures <- list.files(
        path = kegg_plots_dir,
        pattern = "pathview",
        full.names = TRUE
    )
    invisible(lapply(figures, function(figure) {
        cat(paste0("<img src=\"", figure, "\">\n"))
    }))
}
```

```{r footer, child="_footer.Rmd"}
```
